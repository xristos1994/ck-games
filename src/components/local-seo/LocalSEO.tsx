import React, { FC, ReactElement } from 'react';
import { Helmet } from 'react-helmet';

interface IProps {
  title: string;
  canonicalUrl: string;
  description: string;
  keywords: string;
  ogImage: string;
}

const LocalSEO: FC<IProps> = ({
  title,
  canonicalUrl,
  description,
  keywords,
  ogImage
}): ReactElement => {
  return (
    <Helmet>
      {title && <title>{title}</title>}
      {canonicalUrl && <link rel="canonical" href={canonicalUrl} />}
      {title && <meta name="title" content={title} />}
      {description && <meta name="description" content={description} />}
      {keywords && <meta name="keywords" content={keywords} />}

      {canonicalUrl && <meta property="og:url" content={canonicalUrl} />}
      {title && <meta property="og:type" content="website" />}
      {title && <meta property="og:title" content={title} />}
      {description && <meta property="og:description" content={description} />}
      {ogImage && <meta property="og:image" content={ogImage} />}

      {title && <meta name="twitter:card" content="summary" />}
      {description && <meta name="twitter:description" content={description} />}
      {title && <meta name="twitter:title" content={title} />}
      {ogImage && <meta name="twitter:image" content={ogImage} />}
    </Helmet>
  );
};

export { LocalSEO };
