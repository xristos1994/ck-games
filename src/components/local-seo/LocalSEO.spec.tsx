import React from 'react';
import renderer from 'react-test-renderer';
import { LocalSEO } from './LocalSEO';

describe('components/GlobalSEO', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <LocalSEO
        title="title"
        canonicalUrl="canonicalUrl"
        description="description"
        keywords="keywords"
        ogImage="ogImage"
        alternateLangLinks={[
          { langCode: 'EN', link: 'EN_link' },
          { langCode: 'EL', link: 'EL_link' }
        ]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <LocalSEO
        title="title"
        canonicalUrl="canonicalUrl"
        description="description"
        keywords="keywords"
        ogImage="ogImage"
        alternateLangLinks={[]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
