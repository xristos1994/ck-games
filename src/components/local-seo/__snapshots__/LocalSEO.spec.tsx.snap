// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GlobalSEO renders correctly 1`] = `null`;

exports[`components/GlobalSEO renders correctly 2`] = `null`;
